### DISCUSSION TITLE
How can we simulate the effect of the Inner weight GAP?

### DISCUSSION THREAD

**Discussion by matifali** [IBM/aihwkit#425] [Q&A]
There is always a minimum conductance below for RRAM devices that can not be programmed. This creates an Inner Weight Gap (IWG) as described in this [article](https://www.frontiersin.org/articles/10.3389/fnins.2019.00593/full)  How can this be enabled (simulated) while using an RRamPresetDevice in aihwkit?

**Comment by maljoras** [comment]
The devices in AIHWKIT assume a linear mapping of gmin to gmax to wmin and wmax. So it does not really matter what value gmin has it is assumed that one can correct for an offset with a constant reference. Variability in all the parameters can be set, in particular the difference in up versus down update pulse behaviour which indeed does affect accuracy quite dramatically (see e.g. [here](https://aihwkit.readthedocs.io/en/latest/api/aihwkit.simulator.configs.devices.html?highlight=PulsedDevice#aihwkit.simulator.configs.devices.LinearStepDevice) for the parameters underlying the `ReRAMSBPresetDevice` that can be set). Note that `ReRamPresetDevice` was fit to experimental data and should thus in principle model ReRAM qualitatively (at least that ReRAM device that was used for the measurements)
